src/algebra/module/pi.lean

```diff
@@ -105,4 +105,10 @@ instance semimodule' {g : I → Type*} {r : Π i, semiring (f i)} {m : Π i, add
 { add_smul := by { intros, ext1, apply add_smul },
   zero_smul := by { intros, ext1, apply zero_smul } }
 
+instance (α) {r : semiring α} {m : ∀ i, add_comm_monoid $ f i}
```

Comment:
Module/prod could do with this instance too

Reply:
Indeed! My list currently is:

 * product modules
 * submodules, subalgebras
 * algebra R A, if `no_zero_divisors A`